bin/reth/src/args/network_args.rs

```diff
@@ -78,9 +86,19 @@ impl NetworkArgs {
         let chain_bootnodes = chain_spec.chain.bootnodes().unwrap_or_else(mainnet_nodes);
         let peers_file = self.peers_file.clone().unwrap_or(default_peers_file);
 
-        // Configure basic network stack.
+        // Configure peer connections
+        let mut peer_config = config.peers.clone();
+        if let Some(max_inbound_peers) = self.max_inbound_peers {
+            peer_config = peer_config.with_max_inbound(max_inbound_peers);
+        }
+        if let Some(max_inbound_peers) = self.max_outbound_peers {
+            peer_config = peer_config.with_max_outbound(max_inbound_peers);
+        }
```

Comment:
builder pattern with option and non-option values is always a bit horrible -.-

we def want Option values from the CLI, because we need to handle the case were these are set in the reth.toml

I think we should add extra functions to PeerConfig that can handle Option values, just for convenience.

so that this looks a bit nicer.

something like `with_max_inbound_opt(Option<usize>)`